modules/atmosphere/rendering/renderableatmosphere.cpp

```diff
@@ -546,4 +499,78 @@ void RenderableAtmosphere::updateAtmosphereParameters() {
     _deferredcaster->setHardShadows(_hardShadowsEnabled);
 }
 
+// Calculate atmosphere dimming coefficient
+void RenderableAtmosphere::setDimmingCoefficient(const glm::dmat4& modelTransform) {
+    // Calculate if the camera is in the atmosphere and if it is in the sunny region
+    glm::dvec3 cameraPos = global::navigationHandler->camera()->positionVec3();
+    glm::dvec3 planetPos = glm::dvec3(modelTransform * glm::dvec4(0.0, 0.0, 0.0, 1.0));
+    float cameraDistance = static_cast<float>(glm::distance(planetPos, cameraPos));
+    glm::dvec3 normalUnderCamera = glm::normalize(cameraPos - planetPos);
+    glm::dvec3 vecToSun = glm::normalize(-planetPos);
+
+    float cameraSunAngle = glm::degrees(static_cast<float>(
+        glm::acos(glm::dot(vecToSun, normalUnderCamera))
+        ));
+    float sunsetEnd = _atmosphereDimmingSunsetAngle.value().y;
+
+    // If cameraSunAngle is more than 90 degrees, we are in shaded part of globe
+    bool cameraIsInSun = cameraSunAngle <= sunsetEnd;
+    // Atmosphere height is in KM
+    float atmosphereEdge = KM_TO_M * (_planetRadius + _atmosphereHeight);
+    bool cameraIsInAtmosphere = cameraDistance < atmosphereEdge;
+
+    // Don't fade if camera is not in the sunny part of an atmosphere 
+    if (!(cameraIsInAtmosphere && cameraIsInSun)) {
```

Comment:
```suggestion
    if (!cameraIsInAtmosphere || !cameraIsInSun) {
```
Seems easier to read for me